src/scripts/components/ToggleButton.jsx

```diff
@@ -0,0 +1,43 @@
+import cx from 'classnames';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+
+class ToggleButton extends Component {
+  handleClick = () => {
+    if (this.props.onClick) {
+      this.props.onClick(this.props.variable.value, this.props.variable.id);
+    }
+  }
+
+  render() {
+    const { className, onClick, title, type, available, ...opts } = this.props;
+
+    const buttonBaseClass = 'toggle-button';
+    const toggleButtonStyleClasses = {
+      'toggle-button--strikethrough': !available,
+      'button--outline-primary': available,
+
+    };
+
+    const classes = cx(buttonBaseClass, className, {
+      ...toggleButtonStyleClasses,
+    });
+    return (
+      <button type="button" className={classes} onClick={this.handleClick} title={this.props.title} {...opts}>
+        <span className="button__text-wrapper">{this.props.children}</span>
+      </button>
+    );
+  }
+}
+
+ToggleButton.propTypes = {
+  available: PropTypes.bool,
```

Comment:
Can any of these be required or all optional?

Reply:
Can definitely require necessary ones, and I believe I can remove some that are now unnecessary.